10.0/contrib/wfbin/standalone.conf

```diff
@@ -91,9 +86,25 @@ if [ -z "$OPENSHIFT_SMTP_HOST" ]; then
     export OPENSHIFT_SMTP_HOST="localhost"
 fi
 
-max_memory_mb=${OPENSHIFT_GEAR_MEMORY_MB} 
-if [ -z "$OPENSHIFT_GEAR_MEMORY_MB" ]; then
-  max_memory_mb=512
+if [ -z $JAVA_GC_OPTS ]; then
+  JAVA_GC_OPTS="-XX:+UseParallelGC -XX:MaxPermSize=100m -XX:MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90"
+fi
+
+# default to 512meg of total container memory (heap will be a percentage of this value)
+max_memory_mb=512
+# use historical OPENSHIFT_GEAR_MEMORY_MB for container memory size, if specified
+if [ -n "$OPENSHIFT_GEAR_MEMORY_MB" ]; then
+  max_memory_mb=$OPENSHIFT_GEAR_MEMORY_MB
+# otherwise try to set the container memory size based on the cgroup limits
+else
+  CONTAINER_MEMORY_IN_BYTES=`cat /sys/fs/cgroup/memory/memory.limit_in_bytes`
+  DEFAULT_MEMORY_CEILING=$((2**40-1))
+  if [ "${CONTAINER_MEMORY_IN_BYTES}" -lt "${DEFAULT_MEMORY_CEILING}" ]; then
+      max_memory_mb=$((${CONTAINER_MEMORY_IN_BYTES}/1024**2))
+  fi 
+fi
+if [ -z "$JVM_HEAP_RATIO" ]; then
+    JVM_HEAP_RATIO=0.5
```

Comment:
Assuming 64 bit only, are we concerned with the -Xss setting we do (until the 32bit change go in) in the jenkins image?